src/ORM/DataObject.php

```diff
@@ -2228,15 +2236,20 @@ public function getManyManyComponents($componentName, $id = null)
     }
 
     /**
-     * Return the class of a one-to-one component.  If $component is null, return all of the one-to-one components and
-     * their classes. If the selected has_one is a polymorphic field then 'DataObject' will be returned for the type.
+     * Return the class of a all has_one relations.
      *
-     * @return string|array The class of the one-to-one component, or an array of all one-to-one components and
-     *                          their classes.
+     * @return array An array of all has_one components and their classes.
      */
     public function hasOne()
     {
-        return (array)$this->config()->get('has_one');
+        $hasOne = (array) $this->config()->get('has_one');
+        // Boil down has_one spec to just the class name
+        foreach ($hasOne as $relationName => $spec) {
+            if (is_array($spec)) {
+                $hasOne[$relationName] = DataObject::getSchema()->hasOneComponent(static::class, $relationName);
+            }
+        }
```

Comment:
No early returns or breaks here? Can there be multiple and if so, is using the last ok?

Reply:
I'm not sure what you mean?
This will loop through all `has_one` relation declarations, and reduce the new array syntax for any of those relations down to just their class name.

There's nothing to break early for because we have to check each relation.
We're not "using the last" of anything - note we're assigning the same element that we're looking at.

e.g:
```php
private static $has_one = [
    'NormalHasOne' => DataObject::class,
    'FancyHasOne' => [
        'class' => DataObject::class,
        DataObjectSchema::HASONE_IS_MULTIRECIPROCAL => true,
    ],
    'AnotherHasOne' => DataObject::class,
];
```

calling `$record->hasOne()` for a model with the above definition will return an array like so:

```php
[
    'NormalHasOne' => DataObject::class,
    'FancyHasOne' => DataObject::class,
    'AnotherHasOne' => DataObject::class,
]
```